### PR TITLE
Add virtual card request model

### PR DESCRIPTION
**Pull request by hdiniz**
Related: https://github.com/opencollective/opencollective/issues/4312
Related: https://github.com/opencollective/opencollective/issues/6835

Add virtual card requests model and record virtual card request

### CODE REVIEW

**Review by znarf — COMMENTED** [review]
Looks good but let's give it some time before merging.

**Comment by hdiniz** [comment]
Updated with tests and a collection query for the host dashboard.

File: server/graphql/v2/query/collection/VirtualCardRequestsCollectionQuery.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { isEmpty } from 'lodash';
+import { Order } from 'sequelize';
+
+import models from '../../../../models';
+import { GraphQLVirtualCardRequestCollection } from '../../collection/VirtualCardRequestCollection';
+import { GraphQLVirtualCardRequestStatus } from '../../enum/VirtualCardRequestStatus';
+import { fetchAccountWithReference, GraphQLAccountReferenceInput } from '../../input/AccountReferenceInput';
+import { CollectionArgs, CollectionReturnType } from '../../interface/Collection';
+
+const VirtualCardRequestsCollectionQuery = {
```
[the remark this answers is not in the document]
Comment (hdiniz) [reply server/graphql/v2/query/collection/VirtualCardRequestsCollectionQuery.ts:11]: I made the `notes` field private to the collective/host admin with the assumption that the rest of the request could be public (like an expense)

**Review by Betree — APPROVED** [review]
> I made the `notes` field private to the collective/host admin with the assumption that the rest of the request could be public (like an expense)

A virtual card is a payout method, which is (in theory) always private. I don't see any use case yet for exposing the details, but I don't have any blocker as long as we have permission control on private fields.

**Pull request merged**